manager/controllers/app/suite_test.go

```diff
@@ -75,6 +75,17 @@ var _ = BeforeSuite(func(done Done) {
 	if os.Getenv("USE_EXISTING_CONTROLLER") == "true" {
 		logf.Log.Info("Using existing controller in existing cluster...")
 		k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
+		Expect(k8sClient.Create(context.Background(), &v1.ConfigMap{
```

Comment:
This could be part of the environment setup in line 57. Basically another K8s object in the testdata folder that is also loaded. But we can sort the exact testing environment etc out later so this is just a note here. (Then you also would not need the mockup listener)

Reply:
This is very immature. For an existing controller, for example, both cases should be considered (multi/single cluster config). This is an initial version that will make CI pass in a single-cluster environment, but more work should be done to guarantee an effective testing.